Tests/UnitTests/Core/Geometry/TransformStoreTests.cpp

```diff
@@ -0,0 +1,60 @@
+// This file is part of the ACTS project.
+//
+// Copyright (C) 2016 CERN for the benefit of the ACTS project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include <boost/test/unit_test.hpp>
+
+#include "Acts/Definitions/Algebra.hpp"
+#include "Acts/Geometry/GeometryContext.hpp"
+#include "Acts/Geometry/TransformStore.hpp"
+#include "Acts/Surfaces/CylinderSurface.hpp"
+#include "Acts/Surfaces/Surface.hpp"
+
+#include <unordered_map>
+
+namespace Acts::Test {
+
+BOOST_AUTO_TEST_SUITE(Geometry)
+
+BOOST_AUTO_TEST_CASE(GeometryIdMappedTransforms) {
+  std::unordered_map<GeometryIdentifier, Transform3> transformMap;
+
+  auto cylinder0 =
+      Surface::makeShared<CylinderSurface>(Transform3::Identity(), 1.0, 2.0);
+  cylinder0->assignGeometryId(GeometryIdentifier().withVolume(1).withLayer(2));
+  auto contextualTransform0 = Transform3::Identity();
+  contextualTransform0.translation() = Vector3(1.0, 2.0, 3.0);
+  transformMap[cylinder0->geometryId()] = contextualTransform0;
+
+  auto cylinder1 =
+      Surface::makeShared<CylinderSurface>(Transform3::Identity(), 4.0, 5.0);
+  cylinder1->assignGeometryId(GeometryIdentifier().withVolume(7).withLayer(8));
+  auto contextualTransform1 = Transform3::Identity();
+  contextualTransform1.translation() = Vector3(4.0, 5.0, 6.0);
+  transformMap[cylinder1->geometryId()] = contextualTransform1;
```

Comment:
Maybe we can have another example for a "volume" structure in addition to those layers?

Reply:
Hi, this will come. 

In ACTS `volumes` are logical concepts and to this level not yet describing detector volumes.
For the moment, a misaligned barrel volume in the tracker would be misaligned through its surfaces, something we will test with the Generation3 geometry construction (also on DD4hep).

Misaligned volumes such as Muon chambers will be added also later.